src/query/query_manager.c

```diff
@@ -1531,10 +1531,19 @@ xqmgr_execute_query (THREAD_ENTRY * thread_p, const XASL_ID * xasl_id_p, QUERY_I
 	    }
 
 	  /* update the cache entry for the result associated with the used parameter values (DB_VALUE array) if there
-	   * is, or make new one */
-	  list_cache_entry_p =
-	    qfile_update_list_cache_entry (thread_p, &xasl_cache_entry_p->list_ht_no, &params, list_id_p,
-					   xasl_cache_entry_p);
+	   * is, or make new one
+	   * in case list_ht_no is less than 0,
+	   *     the cache is not found and should be newly added (surely list_cache_entry_p is null)
+	   * in case list_ht_no is not less than 0 and list_cache_entry_p is not null
+	   *     the cache entry is found but the entry is used by other transaction
```

Comment:
please add assert for the cases described here in the following if-block

Reply:
the comment is error, should be as follow
in case list_ht_no is not less than 0 and list_cache_entry_p is null

Reply:
list_ht_no >= 0 && list_cache_entry_p != null: the routine doesn't reaches inside of the following if-block

Other cases for list_ht_no and list_cache_entry_p will reach inside of the following if-block.
1) **list_ht_no < 0** && list_cache_entry_p == null
2) list_ht_no >= 0 && **list_cache_entry_p != null**
3) **list_ht_no < 0** && **list_cache_entry_p != null**

The if-condition includes all of the above cases, but only 1) is described in the comment. In order to guarantee that there are no other cases 2) and 3), assert is required.

Reply:
Ok, I add assert for case "list_ht_no < 0 && list_cache_entry_p is not null"